src/components/Menu.jsx

```diff
@@ -1,5 +1,66 @@
+import { getAllTopics } from "../utils/api";
+import { useEffect, useState } from "react";
+import { Loader } from "../components";
+import { Alert } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+
 const Menu = () => {
-  return <></>;
+  const [topics, setTopics] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const navigate = useNavigate();
+
+  const handleHomeClick = () => {
+    navigate('/')
+    window.location.reload(false)
+  }
+
+  const handleTopicClick = (e) => {
+    navigate({pathname:'/',
+    search: `?topic=${e.target.attributes.tag.value}`})
+    window.location.reload(false)
+  }
```

Comment:
This does work but take a look at the notes on useSearchParams for an alternative of this 

https://notes.northcoders.com/courses/js-front-end/react-router#queries-usesearchparams